File: match-bff/internal/domain/robot.go
package domain

import (
	"github.com/robocin/ssl-core/match-bff/pkg/pb/common"
	"github.com/robocin/ssl-core/match-bff/pkg/pb/vision"
)

type Robot struct {
	Confidence      float32   `json:"confidence"`
	RobotId         uint8     `json:"robot_id"`
	RobotColor      string    `json:"robot_color"`
	Position        []float32 `json:"position"`
	Angle           float32   `json:"angle"`
	Velocity        []float32 `json:"velocity"`
	AngularVelocity float32   `json:"angular_velocity"`
	Radius          float32   `json:"radius"`
	Height          float32   `json:"height"`
	DribblerWidth   float32   `json:"dribbler_width"`
}

func NewRobotFromProto(robot *vision.Robot) Robot {
	return Robot{
		Confidence:      robot.Confidence,
		RobotId:         uint8(robot.RobotId.Number),
		RobotColor:      translateColor(robot.RobotId.Color),
		Position:        []float32{robot.Position.X, robot.Position.Y},
		Angle:           robot.Angle,
		Velocity:        []float32{robot.Velocity.X, robot.Velocity.Y},
		AngularVelocity: robot.AngularVelocity,
		Radius:          robot.PhysicalAttributes.Radius,
		Height:          robot.PhysicalAttributes.Height,
		DribblerWidth:   robot.PhysicalAttributes.DribblerWidth,
	}
}

func translateColor(color common.RobotId_Color) string {
	switch color {
	case common.RobotId_COLOR_YELLOW:
		return "yellow"
	case common.RobotId_COLOR_BLUE:
		return "blue"
	}

	return "unspecified"
}

File: player-mfe/src/entities/robot.ts
import { RobotResponse } from "../types/requests";

class Robot {
  private confidence: number;
  private robotId: number;
  private robotColor: RobotColor;
  private position: Vector2;
  private angle: number;
  private velocity: Vector2;
  private angularVelocity: number;
  private radius: number;
  private height: number;
  private dribblerWidth: number;

  constructor(data: RobotResponse) {
    this.confidence = data.confidence;
    this.robotId = data.robot_id;
    this.robotColor = data.robot_color;
    this.position = data.position;
    this.angle = data.angle;
    this.velocity = data.velocity;
    this.angularVelocity = data.angular_velocity;
    this.radius = data.radius;
    this.height = data.height;
    this.dribblerWidth = data.dribbler_width;
  }
}

export default Robot;
